src/Dialect/ONNX/ONNXOps/Tensor/EyeLike.cpp

```diff
@@ -40,24 +40,42 @@ LogicalResult ONNXEyeLikeOpShapeHelper::computeShape() {
 //===----------------------------------------------------------------------===//
 
 //===----------------------------------------------------------------------===//
-// Shape Inference
+// Type Inference
 //===----------------------------------------------------------------------===//
 
-LogicalResult ONNXEyeLikeOp::inferShapes(
-    std::function<void(Region &)> doShapeInference) {
-  if (!hasShapeAndRank(getInput()))
-    return success();
-
-  RankedTensorType inputType = getInput().getType().cast<RankedTensorType>();
-  Type elementType;
+Type ONNXEyeLikeOp::getResultElementType() {
+  const auto inputType = getInput().getType().cast<TensorType>();
```

Comment:
could you use the new casting style?
```suggestion
  const auto inputType = cast<TensorType>(getInput().getType());
```